### PR TITLE
Do not store difficulties in the DB and setup their colors

### PR DESCRIPTION
**Pull request by JulesDT**
I think storing difficulties in a database is not logical for something that is supposed to be very static.

I reworked to make difficulties an enum that can be imported from the different projects.

Alongside this change, I also implemented the colors to follow the image shown on https://github.com/sQuiz-2/Core/issues/12, here's how it looks like now: 
image

I changed a bit of the frontend and this is definitely not my strong suit, so feel free to take a deeper look at this PR and let me know what you think!

### CODE REVIEW

**Comment by Voulto** [comment]
Visually it's almost similar, just that the image has a space with the bottom and right edge.
In comparison, a screen of squiz1:
image

File: packages/api/app/Validators/GameValidator.ts [hunk 1]
```diff
@@ -1,12 +1,13 @@
...
 import { schema, rules } from '@ioc:Adonis/Core/Validator';
+import { DifficultyEnum } from 'App/Enums/Difficulty';
 
 export default class GameValidator {
   constructor(private ctx: HttpContextContract) {}
 
   public schema = schema.create({
     title: schema.string({ trim: true }, [rules.maxLength(80)]),
-    difficultyId: schema.number([rules.exists({ table: 'difficulties', column: 'id' })]),
+    difficultyId: schema.number([rules.range(0, Object.keys(DifficultyEnum).length / 2)]),
```
Comment (M4gie) [inline packages/api/app/Validators/GameValidator.ts:10]: I don't really know why you divide by 2 ? You should only do -1 ?
Also don't forget to change the message `difficultyId.exists` -> `difficultyId.range`

File: packages/api/database/migrations/1603677784030_difficulties.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import BaseSchema from '@ioc:Adonis/Lucid/Schema'
+
```
Comment (M4gie) [inline packages/api/database/migrations/1603677784030_difficulties.ts:2]: This migration will remove the difficulties table but foreign keys in `rounds` and `games` tables will still be there. We don't really have deploy the game in production, so we could maybe just delete `1587805740197_difficulties.ts` and foreign keys ?

**Review by M4gie — COMMENTED** [review]
The database migration will remove the difficulties table but foreign keys in rounds and games tables will still be there. We don't really have deploy the game in production, so we could maybe just delete 1587805740197_difficulties.ts and foreign keys ?

File: packages/api/app/Validators/GameValidator.ts (hunk 1 above)
Comment (JulesDT) [reply packages/api/app/Validators/GameValidator.ts:10]: Yeah the `/ 2` is a mistake

**Comment by JulesDT** [comment]
The `CASCADE` should propagate the deletion of the foreign keys.

If I look at the shcema of the rounds table for example I get
```psql
                                        Table "public.rounds"
    Column     |           Type           | Collation | Nullable |              Default
---------------+--------------------------+-----------+----------+------------------------------------
 id            | integer                  |           | not null | nextval('rounds_id_seq'::regclass)
 question      | character varying(500)   |           |          |
 theme_id      | integer                  |           |          |
 difficulty_id | integer                  |           |          |
 validated     | boolean                  |           | not null | false
 created_at    | timestamp with time zone |           |          |
 updated_at    | timestamp with time zone |           |          |
Indexes:
    "rounds_pkey" PRIMARY KEY, btree (id)
Foreign-key constraints:
    "rounds_theme_id_foreign" FOREIGN KEY (theme_id) REFERENCES themes(id) ON DELETE CASCADE
Referenced by:
    TABLE "answers" CONSTRAINT "answers_round_id_foreign" FOREIGN KEY (round_id) REFERENCES rounds(id) ON DELETE CASCADE
```
Which shows that there's no foreign key to the inexistent table anymore. We could remove the migrations all together but I thought it was nice to keep track of what happened. I'm okay with modifying the old migrations while we're still working on it and it's not live.

Also I'm not sure what the linting error ` 0:0  error  Parsing error: Cannot read property 'map' of undefined` is, it seems unrelated to my changes. Do you know what it's due to?

**Comment by M4gie** [comment]
Ok we will keep a track of the migrations :+1: for the eslint error it's because we need to update our typescript eslint plugin, I create an issue to fix this #17

**Pull request merged**